backend/src/services/verifyToken.js

```diff
@@ -64,8 +64,24 @@ function checkAuthenticated(req, res, next) {
         })
 }
 
+const validateJwtToken = async(req, res, next) => {
+    const token = req.header('jwt-token')
+    if (!token)
+        return res.status(401).json({ message: 'Access denied. No token provided.'})
+    
+    try {
+        const verified = await jwt.verify(token, process.env.JWT_SECRET)
+        req.user = verified
+        next()
+    } catch (err) {
```

Comment:
does jwt.verify() automatically check for expiry date of the token? we just need to make sure to throw descriptive error so we know when the token is expired and when it's just invalid